### ISSUE TITLE
upgrade to golang 1.18

### ISSUE THREAD

**Issue by gengjiawen** [DanSnow/astexplorer-go#2]
RT

**Comment by DanSnow** [comment]
Is there any new syntax in Golang 1.17?

**Comment by TeddyDD** [comment]
There are in 1.18. Bump :)

**Comment by StevenACoffman** [comment]
Already upgraded to Go 1.22

**Comment by StevenACoffman** [comment]
Thank you for contributing this!

@DanSnow it looks like you haven't published the 1.1.0 release to npm https://www.npmjs.com/package/astexplorer-go as 1.0.0 is latest there and in https://registry.yarnpkg.com/astexplorer-go/

Also Go 1.23 allows range over function and iterables.

**Comment by StevenACoffman** [comment]
For reference, this is where this package was added to astexplorer: https://github.com/fkling/astexplorer/pull/459

**Comment by DanSnow** [comment]
Hi @StevenACoffman,

Thanks for reminding me. I have upgraded the go version to 1.23 and released the 1.1.0. Would you like to test it and create a PR on the astexplorer side?

**Comment by StevenACoffman** [comment]
Done! https://github.com/fkling/astexplorer/pull/727

Thanks so much!

**Comment by StevenACoffman** [comment]
@DanSnow Using https://github.com/fkling/astexplorer/pull/727, when I try to run astexplorer locally I get an error:
```
ERROR in ./node_modules/astexplorer-go/parser.wasm
Module not found: Error: Can't resolve 'gojs' in '/Users/steve/Documents/git/astexplorer/website/node_modules/astexplorer-go'
 @ ./node_modules/astexplorer-go/parser.wasm
 @ ./node_modules/astexplorer-go/index.js
 @ ./src/parsers/go/go.js
 @ ./src/parsers sync ^\.\/(?!utils|transpilers)[^/]+\/(transformers\/([^/]+)\/)?(codeExample\.txt|[^/]+?\.js)$
 @ ./src/parsers/index.js
 @ ./src/storage/parse.js
 @ ./src/app.js
```
This is using Node 16 (since that is still max version supported for astexplorer). Steps to reproduce:
```
git clone git@github.com:StevenACoffman/astexplorer.git
cd astexplorer/website
yarn install
yarn build
```

**Comment by StevenACoffman** [comment]
It looks like [when you changed the wasm_exec.js](https://github.com/DanSnow/astexplorer-go/commit/1c35a1445254ed162de0484706aadb50cb3b8b1c#diff-a992dd64a8c626fedc95d613f43602b91f6dc6339893afaf405dc05fa13c9ceaL12)  you didn't add the new
https://github.com/golang/go/blob/go1.23.3/misc/wasm/wasm_exec_node.js file from when they split those into two files.

Dropping the wasm_exec_node.js file into my local astexplorer/website/node_modules/ doesn't fix it though. I'm not sure if `index.js` or `go.js` need to be updated, or the parser.wasm needs to be rebuilt or what.

BTW the Go 1.23.3 uses an identical 
https://github.com/golang/go/blob/go1.23.3/misc/wasm/wasm_exec.js as your current
(and it looks like it's moving for Go 1.24 to 
https://github.com/golang/go/blob/master/lib/wasm/wasm_exec.js anyway)

**Comment by DanSnow** [comment]
@StevenACoffman Thank you for letting me know. I took a look, and I think the current generated wasm can't be used directly by Webpack like before. After testing, I found that manual initialization of wasm is possible, which you can see in my tests at https://github.com/DanSnow/astexplorer-go/blob/master/tests/parser.spec.ts#L21 . However, I'm not sure how Webpack handles it and what the way to make it compatible with Webpack would be.

**Comment by StevenACoffman** [comment]
Yeah! I was able to demonstrate using it directly in both a webpage and from NodeJS: https://github.com/StevenACoffman/astexplorer-go/tree/master/_example

However, I made a number of npm releases on my fork of this repo, and could not get webpack to use it properly, even when I upgraded to latest webpack 5.x

I have a fork of astexplorer with that here https://github.com/StevenACoffman/astexplorer/tree/feat/webpack5

**Comment by DanSnow** [comment]
There is another approach that we might need to modify the Webpack settings on astexplorer's side, allowing us to load the original WASM file using file-loader and initialize it ourselves, similar to what I did in the test https://github.com/DanSnow/astexplorer-go/blob/master/tests/parser.spec.ts#L21

**Comment by StevenACoffman** [comment]
I *really wanted* to do that, but I have no knowledge of webpack and got lost trying to figure out how to better instruct it, or even to debug whatever changes I made.

I noticed that [astexplorer has this webpack config section](https://github.com/fkling/astexplorer/blob/master/website/webpack.config.js#L152-L166) seemed relevant:
```
        // @swc/wasm-web uses a build target assumes to run _without_ bundler, in result
        // contains incompatible syntax to webpack@4 (import.meta.url).
        // in here, augment import.meta with custom loader, also provides path to wasm binary
        // for its initializer to correctly import wasm binary.
        {
          test: /\wasm.js$/,
          include: [path.join(__dirname, 'node_modules', '@swc', 'wasm-web')],
          loader: require.resolve('@open-wc/webpack-import-meta-loader'),
        },
        {
          test: /.wasm$/,
          type: 'asset/resource',
          include: [path.join(__dirname, 'node_modules', '@swc', 'wasm-web')],
        },
```

**Comment by DanSnow** [comment]
The Go version upgrade (the original request) has been completed — the package now runs on Go 1.23 and version 1.1.0 has been published to npm. The remaining webpack compatibility issue in astexplorer itself is tracked upstream at https://github.com/fkling/astexplorer/pull/727. Closing this issue as the original upgrade request is done.
